### PR TITLE
Adding Orchestartion swagger API Specs

### LINKED ISSUE #35: [Backlog] Swagger API specs
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug
/kind feature

**What happened**: New Swagger specs for Orchestration

**What you expected to happen**: Swagger API specs

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Orchestration(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by kumarashit**
**What this PR does / why we need it**: Add Swagger API specs

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #35 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openapi-spec/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+        format: date-time
+        example: 2017-07-10T14:36:58.014Z
+        readOnly: true
+      updatedAt:
+        type: string
+        format: date-time
+        example: 2017-07-10T14:36:58.014Z
+        readOnly: true
+  CustomPropertiesSpec:
+    description: >-
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: remove CustomPropertiesSpec here

File: openapi-spec/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+          - group
+          - workflows
+        properties:
+          name:
+            type: string
+          tenant_id:
+            type: string
+          user_id:
+            type: string
+          group:
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: is this label for service definition?

File: openapi-spec/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+              type: string
+            id:
+              type: string
+            name:
+              type: string
+
+            tenant_id:
+              type: string
+            updated_at:
+              type: string
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: remove updated_at

File: openapi-spec/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+          group:
+            type: string
+          id:
+            type: string
+          name:
+            type: string
+          tenant_id:
+            type: string
+          updated_at:
+            type: string
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: remove updated_at

File: openapi-spec/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+          input:
+            type: string
+            description: "Input provided by the user as part of creating service instance"
+          name:
+            type: string
+          service_id:
+            type: string
+          updated_at:
+            type: string
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: remove updated_at

File: openapi-spec/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+        required:
+          - id
+          - action
+          - parameters
+        properties:
+          id:
+            type: string
+          action:
+            type: string
+            example: "opensds.migration-bucket"
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: is it workflow id or name? please add description

File: openapi-spec/swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+            type: object
+            description: "Input provided by the user as part of creating service instance"
+          name:
+            type: string
+            description: "Name of the service definition of this instance is created"
+          service_id:
+            type: string
+          workflow_source:
+            type: string
+            description: "Name of the action"
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: is it workflow id or name? please add detailed description

File: openapi-spec/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+      - $ref: '#/parameters/tenantId'
+    get:
+      tags:
+        - Service Definitions
+      description: Lists information for all Service Defintions.
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/ServiceDefs'
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: return value should be array

File: openapi-spec/swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+          $ref: '#/responses/HTTPStatus500'
+    post:
+      tags:
+        - Service Definitions
+      description: Registers a Service Definition.
+      parameters:
+        - name: Definition name
+          in: body
+          schema:
+            $ref: '#/definitions/ServiceDefsReq'
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: tenant_id or user_id is not required in request body

File: openapi-spec/swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,534 @@
...
+      - $ref: '#/parameters/tenantId'
+    get:
+      tags:
+        - Service Instance
+      description: Gets all service instances detail.
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/InstanceSpecs'
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: return value should be array

File: openapi-spec/swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,534 @@
...
+          $ref: '#/responses/HTTPStatus500'
+    post:
+      tags:
+        - Service Instance
+      description: Create an instance of service definition.
+      parameters:
+        - name: Instance details
+          in: body
+          schema:
+            $ref: '#/definitions/InstanceReq'
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: id is not required in request body

File: openapi-spec/swagger.yaml [hunk 12]
```diff
@@ -0,0 +1,534 @@
...
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '404':
+          $ref: '#/responses/HTTPStatus404'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+    delete:
+      tags:
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: delete url should be: /v1beta/{tenantId}/orchestration/instances/{instanceId} and there is no request body in delete method

File: openapi-spec/swagger.yaml [hunk 13]
```diff
@@ -0,0 +1,534 @@
...
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '404':
+          $ref: '#/responses/HTTPStatus404'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+  '/v1beta/{tenantId}/orchestration/tasks/{executionId}':
+    parameters:
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: the api looks weird, is it task list api ? I think there two api about task api: list tasks and show a task

File: openapi-spec/swagger.yaml (hunk 9 above)
Comment (kumarashit) [reply openapi-spec/swagger.yaml]: Done

File: openapi-spec/swagger.yaml (hunk 10 above)
Comment (kumarashit) [reply openapi-spec/swagger.yaml]: Done

File: openapi-spec/swagger.yaml (hunk 11 above)
Comment (kumarashit) [reply openapi-spec/swagger.yaml]: Right. Done

File: openapi-spec/swagger.yaml (hunk 12 above)
Comment (kumarashit) [reply openapi-spec/swagger.yaml]: Right. Had misplaced the position in swagger. API behaves accordingly

File: openapi-spec/swagger.yaml (hunk 13 above)
Comment (kumarashit) [reply openapi-spec/swagger.yaml]: This is show execution task

File: openapi-spec/swagger.yaml (hunk 2 above)
Comment (kumarashit) [reply openapi-spec/swagger.yaml]: Service definition

**Pull request merged**